lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetSchemaConverter.java

```diff
@@ -50,18 +50,34 @@
 import static io.trino.spi.type.TinyintType.TINYINT;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
+import static org.apache.parquet.schema.LogicalTypeAnnotation.decimalType;
 import static org.apache.parquet.schema.Type.Repetition.OPTIONAL;
 
 public class ParquetSchemaConverter
 {
+    // Map precision to the number bytes needed for binary conversion.
+    // Based on org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe
+    private static final int[] PRECISION_TO_BYTE_COUNT = new int[39];
```

Comment:
39 can be replaced with `MAX_PRECISION+1`

(the original has the not-so-nice aspect that `PRECISION_TO_BYTE_COUNT` was a wrong name, since at index `p` you had info for precision `p+1`.)